lib/db.js

```diff
@@ -1546,44 +1479,11 @@ var authenticate = function(self, username, password, options, callback) {
  * @param {string} [options.authMechanism=MONGODB-CR] The authentication mechanism to use, GSSAPI, MONGODB-CR, MONGODB-X509, PLAIN
  * @param {Db~resultCallback} [callback] The command result callback
  * @return {Promise} returns Promise if no callback passed
+ * @deprecated This method will no longer be available in the next major release 3.x as MongoDB 3.6 will only allow auth against users in the admin db and will no longer allow multiple credentials on a socket. Please authenticate using MongoClient.connect with auth credentials.
  */
 Db.prototype.authenticate = function(username, password, options, callback) {
-  if(typeof options == 'function') callback = options, options = {};
-  var self = this;
-  // Shallow copy the options
-  options = shallowClone(options);
-
-  // Set default mechanism
-  if(!options.authMechanism) {
-    options.authMechanism = 'DEFAULT';
-  } else if(options.authMechanism != 'GSSAPI'
-    && options.authMechanism != 'DEFAULT'
-    && options.authMechanism != 'MONGODB-CR'
-    && options.authMechanism != 'MONGODB-X509'
-    && options.authMechanism != 'SCRAM-SHA-1'
-    && options.authMechanism != 'PLAIN') {
-      return handleCallback(callback, MongoError.create({message: "only DEFAULT, GSSAPI, PLAIN, MONGODB-X509, SCRAM-SHA-1 or MONGODB-CR is supported by authMechanism", driver:true}));
-  }
-
-  // If we have a callback fallback
-  if(typeof callback == 'function') return authenticate(self, username, password, options, function(err, r) {
-    // Support failed auth method
-    if(err && err.message && err.message.indexOf('saslStart') != -1) err.code = 59;
-    // Reject error
-    if(err) return callback(err, r);
-    callback(null, r);
-  });
-
-  // Return a promise
-  return new this.s.promiseLibrary(function(resolve, reject) {
-    authenticate(self, username, password, options, function(err, r) {
-      // Support failed auth method
-      if(err && err.message && err.message.indexOf('saslStart') != -1) err.code = 59;
-      // Reject error
-      if(err) return reject(err);
-      resolve(r);
-    });
-  });
+  console.warn("Db.prototype.authenticate method will no longer be available in the next major release 3.x as MongoDB 3.6 will only allow auth against users in the admin db and will no longer allow multiple credentials on a socket. Please authenticate using MongoClient.connect with auth credentials.");  
```

Comment:
is it worth adding a deprecation method for other potential deprecations? Something similar to node's [internal deprecate method](https://github.com/nodejs/node/blob/master/lib/internal/util.js#L16-L57)